src/js_native_api_v8.cc

```diff
@@ -576,7 +576,9 @@ Reference::Reference(napi_env env, v8::Local<v8::Value> value, Args&&... args)
     : RefBase(env, std::forward<Args>(args)...),
       _persistent(env->isolate, value),
       _secondPassParameter(new SecondPassCallParameterRef(this)),
-      _secondPassScheduled(false) {
+      _secondPassScheduled(false),
+      _canBeWeak(!env->IsFeatureEnabled(napi_feature_reference_all_types) ||
```

Comment:
We never allowed creating a reference on primitive values (except `Symbol`s). So I believe there is no need to check the feature bit here: reference on primitive values can not be a weak reference.

```suggestion
      _canBeWeak(
```

Reply:
The intent here is to stop offering weak references for Symbols with the new flag and only do it for Objects and Functions to better match the JavaScript spec. But since currently we support weak references for Symbols and have unit tests in `napi_references` for them, I had to put the flag check here. This way the old code works without changes if the flag is not set.

Reply:
I removed the `_canBeWeak` field completely to let V8 engine to decide on the weak references' behavior.
The only true weak references, as it used to be before, are object, external objects, and functions.
All other types are strong references when the ref count is 0.
This behavior can be seen in the new tests.
The only difference between what we used to have before and now is that we allow all types instead of only four (object, external object, function, and symbol) to be ref counted.

It almost feels like that allowing use of `napi_ref` for all value types may not need a special feature, but we should rather just extend the existing behavior as we did previously for Symbols.